.gitlab/.pre/gitlab_configuration.yml

```diff
@@ -22,10 +22,9 @@ test_gitlab_compare_to:
   script:
     - GITLAB_TOKEN=$($CI_PROJECT_DIR/tools/ci/fetch_secret.sh $GITLAB_TOKEN write_api) || exit $?; export GITLAB_TOKEN
     - !reference [.setup_agent_github_app]
-    # Python 3.12 changes default behavior how packages are installed.
-    # In particular, --break-system-packages command line option is
-    # required to use the old behavior or use a virtual env. https://github.com/actions/runner-images/issues/8615
-    - pip install -r tasks/requirements.txt --break-system-packages
```

Comment:
This, for reasons unknown, began failing ([example](https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/827168711)). The issue though is our management of the `setuptools` package [here](https://github.com/DataDog/datadog-agent-buildimages/blob/5b0a495d0c6b799c8627194c41f4e302f0081041/setup_python.sh#L108). Conda already provides that package and installing it via non-Conda means causes a conflict such that pip cannot detect it.

This is a temporary workaround until the very [next PR](https://github.com/DataDog/datadog-agent/pull/33587) which uses dda.